sdks/fabric/interoperation-node-sdk/src/Relay.ts

```diff
@@ -142,9 +145,12 @@ class Relay {
             if (stateError) {
                 throw new Error(`State error: ${stateError}`);
             }
-            if (finalState.getError()) {
+            if (finalState.getStatus() === statePb.RequestState.STATUS.ERROR && finalState.getError()) {
                 throw new Error(`Error from view payload : ${finalState.getError()}`);
             }
+            if (finalState.getStatus() === statePb.RequestState.STATUS.ERROR) {
+                throw new Error(`Error from view payload : UNKNOWN REASON}`);
+            }
```

Comment:
Can you nest the `finalState.getError()` conditional into the first conditional?